src/printer.js

```diff
@@ -338,12 +338,16 @@ function genericPrintNoParens(path, options, print) {
 
     parts.push(" =>");
 
-    return conditionalGroup([
-      concat([ concat(parts), " ", path.call(print, "body") ]),
-      concat([ concat(parts),
-               indent(options.tabWidth,
-                      concat([line, path.call(print, "body")]))])
-    ])
+    const groups = [
+      concat([ concat(parts), " ", path.call(print, "body") ])
+    ];
+    if (n.body.type !== "JSXElement") {
+      groups.push(concat([
+        concat(parts),
+        indent(options.tabWidth, concat([ line, path.call(print, "body") ])),
+      ]));
+    }
+    return conditionalGroup(groups);
```

Comment:
awesome, will do! 

hah, I'd originally wrote it with `body` separated as you have now, and changed it back. Good to hear of the tradeoffs